Lib/blackrenderer/backends/svg.py

```diff
@@ -140,6 +148,21 @@ def _addElement(self, fillPath, fillTransform, paint, gradientTransform):
             (fillPath, fillTransform, clipPath, clipTransform, paint, gradientTransform)
         )
 
+    def _canDrawPath(self, path):
+        if path is None:
+            self._warn(
+                "unbounded",
+                "SVG canvas currently does not support unbounded draw operations",
```

Comment:
I don't know if you want to go that route, but in nanoemoji.colr_to_svg (where we translate from COLR => SVG) we special case a src-in PaintComposite with transparent black PaintSolid background to mean SVG's group-level opacity, since that pattern is so common:

https://github.com/googlefonts/nanoemoji/blob/2ffdd23faad153c5cec763b6e210befc9d704cd1/src/nanoemoji/colr_to_svg.py#L283-L305

Also, I wonder, since your SVGSurface already knows its viewBox, if you could just use that when you find an unbounded paint, as if the unbounded paint were implicitly bounded by a rectangle the size of the viewbox.. though I am not sure when/whether SVG renderers would clip shapes outside the viewbox rectangle, I think it depends on the `overflow` property but IIRC its default changes depending on whether it's applied on the root `<svg>` element of a standalone document or not..

Reply:
The bigger problem is that the SVG backend [doesn't implement composition](https://github.com/BlackFoundryCom/black-renderer/issues/21) to begin with. Once that's there, your second suggestion may indeed help to overcome the problem.